### PR TITLE
[Serving] Support RWKV for serving 

### PR DESCRIPTION
**Pull request by Celve**

### CODE REVIEW

**Comment by tqchen** [comment]
@Celve please double check the ci error, seems there is assertion as of now forcing batch=1

File: python/mlc_llm/serve/engine_base.py [hunk 1]
```diff
@@ -462,6 +513,90 @@ def infer_args_under_mode(
     return kv_cache_config, model_max_single_sequence_length
 
 
+def _infer_kv_cache_config_for_rnn_state(  # pylint: disable=too-many-arguments,too-many-locals,too-many-branches,too-many-statements
```
Comment (MasterJH5574) [inline python/mlc_llm/serve/engine_base.py:516]: Please log out the KV cache config values as in `_infer_kv_cache_config_for_kv_cache`. Here is a pointer PR https://github.com/mlc-ai/mlc-llm/pull/2128

File: cpp/serve/config.cc [hunk 2]
```diff
@@ -292,6 +310,7 @@ String KVCacheConfigNode::AsJSONString() const {
   config["max_total_sequence_length"] =
       picojson::value(static_cast<int64_t>(this->max_total_sequence_length));
   config["prefill_chunk_size"] = picojson::value(static_cast<int64_t>(this->prefill_chunk_size));
+  // TODO!
```
Comment (MasterJH5574) [inline cpp/serve/config.cc:313]: Please address this TODO

File: cpp/serve/config.h [hunk 3]
```diff
@@ -70,13 +70,22 @@ class GenerationConfig : public ObjectRef {
 
 /****************** KV Cache config ******************/
 
+/*! \brief The kind of cache. */
+enum KVStateKind {
+  kNone = 0,
```
Comment (MasterJH5574) [inline cpp/serve/config.h:75]: Ditto.

**Review by MasterJH5574 — COMMENTED** [review]
Thank you @Celve for the great work!! Would be great to have basic test cases to make ensure it can work. Some reference tests here:

* https://github.com/mlc-ai/mlc-llm/blob/main/tests/python/serve/test_serve_engine.py
* https://github.com/mlc-ai/mlc-llm/blob/main/tests/python/serve/server/test_server.py

And also please run these tests above to make sure that this PR does not break the existing functionalities.

**Comment by tqchen** [comment]
@Celve sorry about the latest set of changes, we should stablize now, please do a rebase to resolve conflicts, and once @Hzfengsy confirms it works, we can merge

File: tests/python/serve/server/test_server.py [hunk 4]
```diff
@@ -1296,39 +1296,39 @@ def test_debug_dump_event_trace(
...
-    test_openai_v1_completions_request_cancellation(MODEL, None)
-
-    for msg in CHAT_COMPLETION_MESSAGES:
-        test_openai_v1_chat_completions(MODEL, None, stream=False, messages=msg)
-        test_openai_v1_chat_completions(MODEL, None, stream=True, messages=msg)
-        test_openai_v1_chat_completions_n(MODEL, None, stream=False, messages=msg)
-        test_openai_v1_chat_completions_n(MODEL, None, stream=True, messages=msg)
-        test_openai_v1_chat_completions_openai_package(MODEL, None, stream=False, messages=msg)
-        test_openai_v1_chat_completions_openai_package(MODEL, None, stream=True, messages=msg)
+    # test_openai_v1_models(MODEL, None)
```
Comment (Hzfengsy) [inline tests/python/serve/server/test_server.py:1299]: Revert these changes plz

**Review by Hzfengsy — COMMENTED** [review]
Overall LGTM. cc @MasterJH5574 if you'd like to double check

**Review by Hzfengsy — APPROVED** [review]
